utils/src/main/java/org/wso2/ei/b7a/jms/JmsMessageUtils.java

```diff
@@ -84,15 +83,16 @@ public static boolean isStreamMessage(Message message) {
      *
      * @param message {@link javax.jms.Message} object
      * @return Ballerina array consist of property names
-     * @throws BallerinaJmsException in an error situation
+     * @throws BallerinaException throw a RuntimeException in an error situation
      */
-    public static ArrayValue getPropertyNames(Message message) throws BallerinaJmsException {
+    //TODO: Fix this workaround when Ballerina lang support to return primitive array and error as a union type
+    public static ArrayValue getJmsPropertyNames(Message message) {
         try {
             List<String> propertyNames = Collections.list(message.getPropertyNames());
-            return new ArrayValue(propertyNames.toArray(), new BArrayType(BTypes.typeString));
+            return new ArrayValue(propertyNames.toArray(new String[0]));
 
         } catch (JMSException e) {
-            throw new BallerinaJmsException("Error occurred while getting property names.", e);
+            throw new BallerinaException("Error occurred while getting property names.", e);
```

Comment:
What if we extend BallerinaJmsException from BallerinaException? Rather than using BallerinaException?

Reply:
It'll affect all inter-op calls. Since the fix is temporary, we should revert and throw checked exceptions.

Reply:
+1. Let's create an issue to revert this, pointing to this PR and the Ballerina lang issue

Reply:
Done. https://github.com/wso2/ballerina-integrator/issues/459